### PR TITLE
CI: move prerelease_deps_coverage_64bit_blas to GitHub actions.

### PR DESCRIPTION
**Pull request by tupui**

### CODE REVIEW

**Comment by tupui** [comment]
@rgommers can you have a look? The operations seem to execute as expected.

I had to remove building with no warnings and I also skipped a test which made the job either hanging forever or failing with nans (In linprog `test_bug_10466`). This might be the same thing happening on Azure. What should I do here?

Otherwise it's going through and coverage is being uploaded. There are some warnings there, but not sure how valid they are.

File: scipy/optimize/tests/test_linprog.py [hunk 1]
```diff
@@ -1648,6 +1648,7 @@ def test_bug_10466(self):
         """
         Test that autoscale fixes poorly-scaled problem
         """
+        pytest.skip()
```
Comment (rgommers) [inline scipy/optimize/tests/test_linprog.py]: The standard way of doing this is as a decorator, and using `reason=`.

**Comment by rgommers** [comment]
> I had to remove building with no warnings

I don't see that removed here? This job also didn't have it; we only have one job with `-Werror` which is Meson on Linux I believe.

> hanging forever or failing with nans (In linprog `test_bug_10466`). This might be the same thing happening on Azure. What should I do here?

How about reopening the relevant issue (gh-10466), skipping it for now, and asking the `linprog` experts to take it from there?

File: scipy/optimize/tests/test_linprog.py (hunk 1 above)
Comment (tupui) [reply scipy/optimize/tests/test_linprog.py]: Sorry, I was not clear here. I am not sure this should be done at all. I committed this to be sure it was the reason CI was not going. There is already a skipif so if we do want to skip this entirely, then I would rewrote this one.

**Comment by tupui** [comment]
> > I had to remove building with no warnings
> 
> I don't see that removed here? This job also didn't have it; we only have one job with `-Werror` which is Meson on Linux I believe.

Ok then good. I was not sure if we wanted to use this all the time or not.

> > hanging forever or failing with nans (In linprog `test_bug_10466`). This might be the same thing happening on Azure. What should I do here?
> 
> How about reopening the relevant issue ([scipygh-10466](https://github.com/scipy/scipy/issues/10466)), skipping it for now, and asking the `linprog` experts to take it from there?

Ok, I will do this then 👍 

I will update this and push the changes to the PR on SciPy. Thank you Ralf 😃

**Comment by rgommers** [comment]
> Ok then good. I was not sure if we wanted to use this all the time or not.

No, that's too fragile. We _may_ want to have it on another 1 or 2 configs, but so far only GCC on Linux with Meson builds without warnings, everything else will just break.

Such a change definitely shouldn't be mixed with migration the job I'd say.

**Comment by rgommers** [comment]
Best new candidate for a `-Werror` job would be Meson macOS job with Clang (because Clang is more strict).

**Pull request closed without merging**